src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx

```diff
@@ -0,0 +1,211 @@
+}: AccordionAssignmentSheetProps) => {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const [isChecked, setIsChecked] = useState<{ [taskId: string]: boolean }>({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentTask, setCurrentTask] = useState<{
+    title: string;
+    description: string;
+  }>({ title: '', description: '' });
+
+  const handleCheckboxChange = (taskId: string) => {
+    setIsChecked((prevChecked) => ({
+      ...prevChecked,
+      [taskId]: !prevChecked[taskId],
+    }));
+  };
+
+  const handleAllCheckedChange = (categoryId: string) => {
+    // Get all task IDs for the category
+    const taskIds = Object.values(rotationData.rotations).flatMap(
+      (rotation) =>
+        rotation.categories
+          .find((category) => category.id === categoryId)
+          ?.tasks.map((task) => task.id) || [],
+    );
+
+    // Check if all tasks in the category are checked
+    const allChecked = taskIds.every((taskId) => isChecked[taskId]);
+
+    // Create a new checked state with toggled values
+    const newChecked = { ...isChecked };
+    taskIds.forEach((taskId) => {
+      newChecked[taskId] = !allChecked;
+    });
+
+    // Update the checked state
+    setIsChecked(newChecked);
+  };
+
+  const { handleSubmit } = useForm();
+
+  const handleSaveClick = () => {
+    const updatedTasks = Object.values(rotationData.rotations)
+      .flatMap((rotation) =>
+        rotation.categories.flatMap((category) => category.tasks),
+      )
+      .filter((task) => isChecked[task.id])
+      .map((task) => ({
+        ...task,
+        isCompleted: isChecked[task.id] || false,
+      }));
+    console.log('updated', updatedTasks);
+  };
+
+  // TODO: Implement the onSubmit click functionality
+  const onSubmit = async () => {
+    setIsLoading(true);
+
+    // Submit the form data
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    setTimeout(() => {
+      setIsLoading(false);
+      toast({
+        variant: 'default',
+        description: 'Updated successfully!',
+      });
+    }, 1000);
+  };
+
+  return (
+    <>
+      {isLoading && <LoadingSpinner isLoading={true} />}
+      <Accordion
+        type="single"
+        collapsible
+        className="w-full"
+        defaultValue={`item-1`}
+      >
+        <form onSubmit={handleSubmit(onSubmit)}>
+          {Object.entries(rotationData.rotations).map(([key, rotation]) => {
+            const numericKey = parseInt(key);
+            if (numericKey > 2) {
+              return null;
+            }
+            return (
+              <AccordionItem value={`item-${key}`} key={key}>
+                <AccordionTrigger>
+                  {rotation.startDate} - {rotation.endDate}
+                </AccordionTrigger>
+                <AccordionContent
+                  className={'bg-primary-lightest p-0 rounded-none'}
+                  asChild
+                >
+                  {rotation.categories.map((category) => (
```

Comment:
If possible, could you make it so that 'No task' is displayed when the value of `category` is null?